{
  try {
    final JobGraph jobGraph=getJobGraph(plan);
    executor.submitJobDetached(jobGraph);
    awaitRunning(executor.getLeaderGateway(TestingUtils.TESTING_DURATION()),jobGraph.getJobID(),TestingUtils.TESTING_DURATION());
    Thread.sleep(msecsTillCanceling);
    cancelJob(executor.getLeaderGateway(TestingUtils.TESTING_DURATION()),jobGraph.getJobID(),new FiniteDuration(maxTimeTillCanceled,TimeUnit.MILLISECONDS));
    JobManagerActorTestUtils.waitForJobStatus(jobGraph.getJobID(),JobStatus.CANCELED,executor.getLeaderGateway(TestingUtils.TESTING_DURATION()),TestingUtils.TESTING_DURATION());
  }
 catch (  Exception e) {
    LOG.error("Exception found in runAndCancelJob.",e);
    e.printStackTrace();
    Assert.fail(e.getMessage());
  }
}
